src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -39,7 +40,9 @@ object BoringUtils {
           def toFirrtl = SourceAnnotation(component.toNamed, name)
           def transformClass = classOf[WiringTransform] },
         new ChiselAnnotation {
-          def toFirrtl = DontTouchAnnotation(component.toNamed) })
+          def toFirrtl = DontTouchAnnotation(component.toNamed) },
+        new ChiselAnnotation {
+          def toFirrtl = NoDedupAnnotation(component.toNamed.module) })
```

Comment:
I would think we would still want deduplication for Modules that all have the same boring annotations right? Given that we don't really have a way to distinguish NoDedup to YouCanDedupSomeOfUsTogether though, I think that's follow on work

Reply:
I modified the test case (just now) to show what I perceive as the issue with needing deduplication. If you have two instances of a source and you only want to wire one, then without deduplication they look the same. That manifests as a `WiringException` complaining that a source to sink mapping can't be determined due to multiple sources being equidistant to the same sink. The same problem exists for sinks. You wind up wiring all modules as opposed to just the one you want.

The use of `addSource`/`addSink` doesn't manifest that way typically from a usability perspective (though it still can). A user would have to mixin a trait to a module or do some `{ ... }` stuff when constructing the module (which most people won't do). This only occurred to me when I started playing with your `bore` object example.

Any better way to handle this? (Other than instance annotations...) Am I seeing a non-issue here?